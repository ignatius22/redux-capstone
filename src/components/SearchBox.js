import React from 'react';
import PropTypes from 'prop-types';

const SearchBox = ({ searchChange }) => (
  <div className="pa2">
    <input
      data-testid="input"
      className="pa3 ba b--green bg-lightest-blue"
      type="search"
      placeholder="search your chicken"
      onChange={searchChange}
    />
  </div>
);

SearchBox.propTypes = {
  searchChange: PropTypes.func.isRequired,
};

export default SearchBox;
