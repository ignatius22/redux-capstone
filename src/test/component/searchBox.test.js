/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import 'jest';
import SearchBox from '../../components/SearchBox';
import ReactDOM from 'react-dom'



afterEach(cleanup);

it('* renders *', () => {
  const { asFragment } = render(<SearchBox onChange="any props" />);
  expect(asFragment()).toMatchSnapshot();
});

it('* renders without crashing *', () => {
	const div = document.createElement('div')
	ReactDOM.render(<SearchBox />, div);
})

describe("input value", () => {
	it("return update onchange", () => {
		const { queryByPlaceholderText } = render(<SearchBox  />)
		const searchInput = queryByPlaceholderText('search chicken');
		fireEvent.change(searchInput, { target: { value: "test" } })
		expect(searchInput.value).toBe("test")
		
	})
})