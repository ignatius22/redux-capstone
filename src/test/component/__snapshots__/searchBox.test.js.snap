// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`* renders * 1`] = `
<DocumentFragment>
  <div
    class="pa2"
  >
    <input
      class="pa3 ba b--green bg-lightest-blue"
      data-testid="input"
      placeholder="search your chicken"
      type="search"
    />
  </div>
</DocumentFragment>
`;
